### ISSUE TITLE
Feat request: Ability to specify default accessibility

### ISSUE THREAD

**Issue by trusktr** [microsoft/TypeScript#35452]
## Search Terms

typescript option default accessibility

## Suggestion

It'd be neat to be able to configure the default accessibility of a class member. For example, if the option is set to `private`, then any members without accessibility explicitly defined would be `private`.

I like the idea of being require to explicitly defining what is public, and what is protected, with everything (except maybe the constructor) defaulting to private. It'll be a stricter mode that forces people to think about what API they expose to others, and also prevents bugs from methods accidentally not being called because the default public accessibility doesn't catch unused vars/functions.

## Use Cases

Basically prevent errors, and forces people to explicitly define what APIs are meant to be exposed, and what parts are internal implementation.

## Examples

Just imagine an option (f.e. `defaultAccessibility` or similar name) is set to `private`, and that writing the following results in an error:

```ts
class One {one() {}}
const o = new One
o.one() // ERROR, one is private
```

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by RyanCavanaugh** [comment]
We generally don't add flags that radically change the interpretation of a given block of code.
